### PR TITLE
Allow disabling TestRunner without Panic

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description
Fix testrunner deployment

# Context
## What is a TestRunner?
A TestRunner is a testing utility used exclusively by Gloo Edge. (this utility should really live in Gloo Edge, but we won't do that for now). In addition to installing gloo, we optionally allow developers to install the TestRunner, a simple HttpEcho container that can be used as an endpoint for requests. 

## How is it used?
https://github.com/solo-io/gloo/blob/master/test/kube2e/gateway/gateway_suite_test.go#L69 In each of our tests, we use this SoloTestHelper to install gloo, and optionally deploy this testRunner container: https://github.com/solo-io/k8s-utils/blob/master/testutils/helper/install.go#L106.

## What is the problem?
When you disable the TestRunner, and then InstallGloo, you hit this line of code: https://github.com/solo-io/k8s-utils/blob/master/testutils/helper/install.go#L171

The issue is that the interface is not nil, so the conditional passes, but the interface implementation is nil, so it panics as soon as it hits: https://github.com/solo-io/k8s-utils/blob/master/testutils/helper/testrunner.go#L65

We never hit this before because all tests deploy the test runner.

Example stack trace:
```
[AfterSuite] AfterSuite
/Users/samheilbron/go/src/github.com/solo-io/solo-projects/test/federation-kube2e/canary/canary_suite_test.go:77

  Test Panicked
  runtime error: invalid memory address or nil pointer dereference
  /Users/samheilbron/.gvm/gos/go1.18.1/src/runtime/panic.go:844

  Full Stack Trace
  panic({0x81cc920, 0xb86f6d0})
        /Users/samheilbron/.gvm/gos/go1.18.1/src/runtime/panic.go:844 +0x258
  github.com/solo-io/k8s-utils/testutils/helper.(*testRunner).Terminate(0x0)
        <autogenerated>:1 +0x31
  github.com/solo-io/k8s-utils/testutils/helper.(*SoloTestHelper).uninstallGloo(0xc001410eb8, 0x0)
        /Users/samheilbron/go/src/github.com/solo-io/k8s-utils/testutils/helper/install.go:220 +0xdc

```

## Why should we fix this?
This utility exposes an API to disable the TestRunner and we should support that.

Also, in Gloo Federation, we only need to install Gloo without hte test runner. Furthermore, the installation of the TestRunner seems like it should be decoupled from the installation of Gloo so you can keep the lifecycles separate.

### CODE REVIEW

File: testutils/helper/install.go [hunk 1]
```diff
@@ -101,20 +101,20 @@ func NewSoloTestHelper(configFunc TestConfigFunc) (*SoloTestHelper, error) {
 		testConfig.InstallNamespace = testConfig.version
 	}
 
+	testHelper := &SoloTestHelper{
+		TestConfig: &testConfig,
+	}
+
 	// Optionally, initialize a test runner
-	var testRunner *testRunner
 	if testConfig.DeployTestRunner {
```
Comment (jenshu) [inline testutils/helper/install.go:109]: what is a test runner used for, and in what scenarios would we want / not want to use it?

Comment (elcasteel) [reply testutils/helper/install.go:109]: same question/do you have some more context for this change?

Comment (sam-heilbron) [reply testutils/helper/install.go:109]: heh yeah tossed this up but didn't have time to add hte context, will fill in now

**Pull request merged**